src/pipecat/frames/frames.py

```diff
@@ -438,6 +438,20 @@ class OutputDTMFFrame(DTMFFrame, DataFrame):
 #
 
 
+@dataclass
+class InterruptionConfig:
+    """Configuration for interruption behavior.
+
+    When specified, the bot will not be interrupted immediately when the user speaks.
+    Instead, interruption will only occur when the configured conditions are met.
+
+    Args:
+        min_words: If set, user must speak at least this many words to interrupt
+    """
+
+    min_words: Optional[int] = None
```

Comment:
Maybe we could have an empty base class `InterruptionStrategy` and then subclasses, for example:

```
@dataclass
class InterruptionStrategy:
    pass

@dataclass
class NumWordsInterruptionStrategy(InterruptionStrategy):
    """Strategy for interruption behavior.
    When specified, the bot will not be interrupted immediately when the user speaks.
    Instead, interruption will only occur when the configured conditions are met.
    Args:
        min_words: If set, user must speak at least this many words to interrupt
    """

    min_words: Optional[int] = None
```

we would then use `isinstance(config, NumWordsInterruptionStrategy`. The Strategy indicates there's going to be a different behavior depending on the class you choose.